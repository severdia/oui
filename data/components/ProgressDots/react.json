{"description":"Generates an `ProgressDots`","displayName":"ProgressDots","methods":[],"props":{"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"}},"private":false}