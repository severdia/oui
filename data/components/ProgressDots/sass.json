{"name":"Progress Dots\n","description":"Creates a row of three animated progress dots.","start":{"line":1,"column":1},"end":{"line":10,"column":4},"example":[{"type":"html","description":null,"code":"<ul class=\"#{$namespace}progress-dots\">\n  <li></li>\n  <li></li>\n  <li></li>\n</ul>"}]}