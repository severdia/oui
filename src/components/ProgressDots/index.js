import React from 'react';
import PropTypes from 'prop-types';

/**
 * Generates an `ProgressDots`
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const ProgressDots = ({ testSection }) => {
  return (
    <ul
      data-oui-component={ true }
      data-test-section={ testSection }
      className="oui-progress-dots">
      <li></li>
      <li></li>
      <li></li>
    </ul>
  );
};

ProgressDots.propTypes = {
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
};

ProgressDots.displayName = 'ProgressDots';

export default ProgressDots;
