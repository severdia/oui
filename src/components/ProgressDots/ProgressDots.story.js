import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';

import ProgressDots from './index.js';

const stories = storiesOf('ProgressDots', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories.add('Default', (() => {
  return <ProgressDots testSection="test-progress-dots" />;
}));
