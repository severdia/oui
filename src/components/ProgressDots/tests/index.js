import React from 'react';
import ProgressDots from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/ProgressDots', () => {
  it('should render correctly without props', () => {
    const output = shallow(
      <ProgressDots />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly with testSection prop', () => {
    const output = shallow(
      <ProgressDots testSection='testing-foo' />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
