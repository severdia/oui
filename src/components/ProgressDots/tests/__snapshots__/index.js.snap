// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ProgressDots should render correctly with testSection prop 1`] = `
<ul
  className="oui-progress-dots"
  data-oui-component={true}
  data-test-section="testing-foo"
>
  <li />
  <li />
  <li />
</ul>
`;

exports[`components/ProgressDots should render correctly without props 1`] = `
<ul
  className="oui-progress-dots"
  data-oui-component={true}
>
  <li />
  <li />
  <li />
</ul>
`;
